syntax = "proto2";

package message;

message Message {
    optional string user = 1;
    optional string function = 2;
    optional int32 idx = 3;

    enum MessageType {
        BIND = 1;
        CALL = 2;
        KILL = 3;
    }

    optional MessageType type = 4 [default = CALL];

    optional string resultKey = 5;
    optional string statusKey = 6;
    optional bool success = 7;
    optional bool isAsync = 8;
    optional bool isPython = 9;

    optional bytes inputData = 10;
    optional bytes outputData = 11;
}

